Source/Bogus/DataSet.cs

```diff
@@ -11,68 +11,76 @@ namespace Bogus
    public class DataSet : ILocaleAware, IHasRandomizer
    {
       /// <summary>
-      /// Resolves the 'category' type of a dataset type; respects the 'DataCategory' attribute.
-      /// </summary>
-      public static string ResolveCategory(Type type)
-      {
-         var categoryAttribute = type.GetCustomAttributeX<DataCategoryAttribute>();
-         return categoryAttribute != null ? categoryAttribute.Name : type.Name.ToLowerInvariant();
-      }
-
-      /// <summary>
-      /// Default constructor
+      /// Initializes a new instance of the <see cref="DataSet"/> class.
       /// </summary>
+      /// <param name="locale">The locale wanting to be set. Default is "en" for English.</param>
+      /// <exception cref="BogusException">
+      /// When the given <paramref name="locale"/> isn't found.
+      /// </exception>
       public DataSet(string locale = "en")
       {
-         if( !Database.LocaleResourceExists(locale) )
+         if (!Database.LocaleResourceExists(locale))
+         {
             throw new BogusException(
-               $"The locale '{locale}' does not exist. To see all available locales visit {AssemblyVersionInformation.AssemblyDescription}."
-               );
+               $"The locale '{locale}' does not exist. To see all available locales visit {AssemblyVersionInformation.AssemblyDescription}.");
+         }
 
          this.Locale = locale;
 
          this.Category = ResolveCategory(this.GetType());
       }
 
       /// <summary>
-      /// See <see cref="SeedNotifier"/>
+      /// Gets or sets the category name inside the locale.
+      /// </summary>
+      protected string Category { get; set; }
+
+      /// <summary>
+      /// Gets or sets the current locale of the data set.
+      /// </summary>
+      public string Locale { get; set; }
+
+      /// <summary>
+      /// See <see cref="SeedNotifier"/>.
       /// </summary>
       protected SeedNotifier Notifier = new SeedNotifier();
 
       private Randomizer randomizer;
 
       /// <summary>
-      /// The Randomizer
+      /// Gets or sets the random used to generate values.
       /// </summary>
       public Randomizer Random
       {
-         get => this.randomizer ?? (this.Random = new Randomizer());
+         get => this.randomizer ?? (this.randomizer = new Randomizer());
```

Comment:
Hey Bart,

I think only setting `this.randomizer` variable is a breaking change.

Here's why: I think this should be reverted back to `this.Random` because `this.Random` causes the set method to execute 1) setting `this.randomzier` and 2) firing `this.Notifier.Notify(value)` to notify all dependency datasets that the parent randomizer has changed.

So, this change essentially breaks the notification mechanism which is critical for deterministic behavior. You can read more about these internals here: https://github.com/bchavez/Bogus/pull/101

Reply:
Yup that was my mistake and stupid if I'm honest. Reverted the change.